### PR TITLE
Uni prop

### PR DESCRIPTION
**Pull request by gxingcvx**
Make appropriate modifications to scale the propagators to ensure: (1) the nonlinear propagators are consistent with the analytical solutions; (2) the linear propagators (Born modeling) are consistent with the nonlinear propagators.

### CODE REVIEW

File: src/jop_prop2DAcoIsoDenQ_DEO2_FDTD.jl [hunk 1]
```diff
@@ -513,6 +518,11 @@ function JopProp2DAcoIsoDenQ_DEO2_FDTD_nonlinearforward!(d::AbstractArray, m::Ab
     set_zero_subnormals(false)
     JopProp2DAcoIsoDenQ_DEO2_FDTD_stats(kwargs[:stats], kwargs[:ginsu], ntmod_wav, time()-time1, cumtime_io, cumtime_ex)
 
+    if length(d) > 0
+        d .= 0
```
Comment (samtkaplan) [inline src/jop_prop2DAcoIsoDenQ_DEO2_FDTD.jl:522]: please double check for me, but it looks like `d .= 0` already happens inside of the `interpforward!` method, so no need to do this here.  Same goes for all other new applications of interforward! as well.

File: src/jop_prop2DAcoIsoDenQ_DEO2_FDTD.jl [hunk 2]
```diff
@@ -806,7 +820,7 @@ function JopProp2DAcoIsoDenQ_DEO2_FDTD_df′!(δm::AbstractArray, δd::AbstractA
     end
 
     free(p)
-
+
```
Comment (samtkaplan) [inline src/jop_prop2DAcoIsoDenQ_DEO2_FDTD.jl:823]: looks like some extra while space.  please delete.  thanks.

**Review by samtkaplan — CHANGES_REQUESTED** [review]
Thanks @gxingcvx.  This looks great.   Please address the 2 comments.

**Review by samtkaplan — APPROVED** [review]

**Review by jkwashbourne-oss — APPROVED** [review]
Looks great Guangchi, thanks for this!

**Pull request merged**